ethportal-api/src/types/execution/transaction.rs

```diff
@@ -54,6 +54,32 @@ impl Transaction {
     }
 }
 
+impl Encodable for Transaction {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        match self {
+            Self::Legacy(tx) => tx.rlp_append(s),
+            Self::AccessList(tx) => tx.rlp_append(s),
+            Self::EIP1559(tx) => tx.rlp_append(s),
+        }
+    }
+}
+
+impl Decodable for Transaction {
```

Comment:
It really isn't useful / ideal that we don't use the native `Decodable`/`Encodable` traits for rlp stuff on `Transaction`. I'm working on replacing our custom rlp methods on `Transaction` in another pr, but decided to leave those changes out of this pr to avoid bloat.